In [70]:
import numpy as np
import pandas as pd
pd.set_option('display.max_columns', 500)

In [2]:
train_provider = pd.read_csv('Train-1542865627584.csv')

In [3]:
train_provider.head(3)

Provider PotentialFraud
0  PRV51001             No
1  PRV51003            Yes
2  PRV51004             No

In [4]:
#to get the number of fradulant providers
train_provider.PotentialFraud.value_counts()

No     4904
Yes     506
Name: PotentialFraud, dtype: int64

In [5]:
#percentage of fradulant data
(506/4904)*100

10.318107667210441

In [27]:
train_provider.shape

(5410, 2)

In [9]:
test_provider = pd.read_csv('Test-1542969243754.csv')

In [10]:
test_provider.head(3)

Provider
0  PRV51002
1  PRV51006
2  PRV51009

In [28]:
test_provider.shape

(1353, 1)

In [11]:
train_bene = pd.read_csv('Train_Beneficiarydata-1542865627584.csv')

In [12]:
train_bene.head(3)

BeneID         DOB  DOD  Gender  Race RenalDiseaseIndicator  State  \
0  BENE11001  1943-01-01  NaN       1     1                     0     39   
1  BENE11002  1936-09-01  NaN       2     1                     0     39   
2  BENE11003  1936-08-01  NaN       1     1                     0     52   

   County  NoOfMonths_PartACov  NoOfMonths_PartBCov  ...  \
0     230                   12                   12  ...   
1     280                   12                   12  ...   
2     590                   12                   12  ...   

   ChronicCond_Depression  ChronicCond_Diabetes  ChronicCond_IschemicHeart  \
0                       1                     1                          1   
1                       2                     2                          2   
2                       2                     2                          1   

   ChronicCond_Osteoporasis  ChronicCond_rheumatoidarthritis  \
0                         2                                1   
1                         2                                2   
2                         2                                2   

   ChronicCond_stroke  IPAnnualReimbursementAmt  IPAnnualDeductibleAmt  \
0                   1                     36000                   3204   
1                   2                         0                      0   
2                   2                         0                      0   

   OPAnnualReimbursementAmt  OPAnnualDeductibleAmt  
0                        60                     70  
1                        30                     50  
2                        90                     40  

[3 rows x 25 columns]

In [13]:
train_bene.columns

Index(['BeneID', 'DOB', 'DOD', 'Gender', 'Race', 'RenalDiseaseIndicator',
       'State', 'County', 'NoOfMonths_PartACov', 'NoOfMonths_PartBCov',
       'ChronicCond_Alzheimer', 'ChronicCond_Heartfailure',
       'ChronicCond_KidneyDisease', 'ChronicCond_Cancer',
       'ChronicCond_ObstrPulmonary', 'ChronicCond_Depression',
       'ChronicCond_Diabetes', 'ChronicCond_IschemicHeart',
       'ChronicCond_Osteoporasis', 'ChronicCond_rheumatoidarthritis',
       'ChronicCond_stroke', 'IPAnnualReimbursementAmt',
       'IPAnnualDeductibleAmt', 'OPAnnualReimbursementAmt',
       'OPAnnualDeductibleAmt'],
      dtype='object')

In [14]:
len(train_bene.columns)

25

In [29]:
train_bene.shape

(138556, 25)

In [15]:
train_inpatient = pd.read_csv('Train_Inpatientdata-1542865627584.csv')

In [16]:
train_inpatient.head(3)

BeneID   ClaimID ClaimStartDt  ClaimEndDt  Provider  \
0  BENE11001  CLM46614   2009-04-12  2009-04-18  PRV55912   
1  BENE11001  CLM66048   2009-08-31  2009-09-02  PRV55907   
2  BENE11001  CLM68358   2009-09-17  2009-09-20  PRV56046   

   InscClaimAmtReimbursed AttendingPhysician OperatingPhysician  \
0                   26000          PHY390922                NaN   
1                    5000          PHY318495          PHY318495   
2                    5000          PHY372395                NaN   

  OtherPhysician AdmissionDt  ... ClmDiagnosisCode_7  ClmDiagnosisCode_8  \
0            NaN  2009-04-12  ...               2724               19889   
1            NaN  2009-08-31  ...                NaN                 NaN   
2      PHY324689  2009-09-17  ...                NaN                 NaN   

  ClmDiagnosisCode_9 ClmDiagnosisCode_10 ClmProcedureCode_1  \
0               5849                 NaN                NaN   
1                NaN                 NaN             7092.0   
2                NaN                 NaN                NaN   

  ClmProcedureCode_2 ClmProcedureCode_3 ClmProcedureCode_4 ClmProcedureCode_5  \
0                NaN                NaN                NaN                NaN   
1                NaN                NaN                NaN                NaN   
2                NaN                NaN                NaN                NaN   

  ClmProcedureCode_6  
0                NaN  
1                NaN  
2                NaN  

[3 rows x 30 columns]

In [18]:
len(train_inpatient.columns)

30

In [30]:
train_inpatient.shape

(40474, 30)

In [19]:
train_inpatient.columns

Index(['BeneID', 'ClaimID', 'ClaimStartDt', 'ClaimEndDt', 'Provider',
       'InscClaimAmtReimbursed', 'AttendingPhysician', 'OperatingPhysician',
       'OtherPhysician', 'AdmissionDt', 'ClmAdmitDiagnosisCode',
       'DeductibleAmtPaid', 'DischargeDt', 'DiagnosisGroupCode',
       'ClmDiagnosisCode_1', 'ClmDiagnosisCode_2', 'ClmDiagnosisCode_3',
       'ClmDiagnosisCode_4', 'ClmDiagnosisCode_5', 'ClmDiagnosisCode_6',
       'ClmDiagnosisCode_7', 'ClmDiagnosisCode_8', 'ClmDiagnosisCode_9',
       'ClmDiagnosisCode_10', 'ClmProcedureCode_1', 'ClmProcedureCode_2',
       'ClmProcedureCode_3', 'ClmProcedureCode_4', 'ClmProcedureCode_5',
       'ClmProcedureCode_6'],
      dtype='object')

In [21]:
train_outpatient = pd.read_csv('Train_Outpatientdata-1542865627584.csv')

In [23]:
train_outpatient.columns

Index(['BeneID', 'ClaimID', 'ClaimStartDt', 'ClaimEndDt', 'Provider',
       'InscClaimAmtReimbursed', 'AttendingPhysician', 'OperatingPhysician',
       'OtherPhysician', 'ClmDiagnosisCode_1', 'ClmDiagnosisCode_2',
       'ClmDiagnosisCode_3', 'ClmDiagnosisCode_4', 'ClmDiagnosisCode_5',
       'ClmDiagnosisCode_6', 'ClmDiagnosisCode_7', 'ClmDiagnosisCode_8',
       'ClmDiagnosisCode_9', 'ClmDiagnosisCode_10', 'ClmProcedureCode_1',
       'ClmProcedureCode_2', 'ClmProcedureCode_3', 'ClmProcedureCode_4',
       'ClmProcedureCode_5', 'ClmProcedureCode_6', 'DeductibleAmtPaid',
       'ClmAdmitDiagnosisCode'],
      dtype='object')

In [31]:
train_outpatient.shape

(517737, 27)

In [24]:
test_bene = pd.read_csv('Test_Beneficiarydata-1542969243754.csv')

In [32]:
test_bene.shape

(63968, 25)

In [25]:
test_inpatient = pd.read_csv('Test_Inpatientdata-1542969243754.csv')

In [33]:
test_inpatient.shape

(9551, 30)

In [26]:
test_outpatient = pd.read_csv('Test_Outpatientdata-1542969243754.csv')

In [34]:
test_outpatient.shape

(125841, 27)

### Warmup Question 1:
- How many medical doctors are there in the train outpatient dataset?
- How many medical doctors are there in the train inpatient dataset? 
Do they match with those from the outpatient record?
- Do those inpatient patients show worse health conditions (in terms of chronic diseases) than typical
outpatient patients, or do those who have more visits to the providers have worse health conditions?
Provide an analysis on these issues.
- How many distinct patients (**BeneIDs**) are there in the in/out-patient datasets? 
- Do doctors serve for different providers? Study the distribution of hospital counts/doctor? Is it possible to
characterize those doctors who move around among different providers?
- Do patients go to different hospitals? Study the distribution of hospital counts/patient? It is possible to characterize
those patients who receive services from a lot of different hospitals?
- Do the same providers provide both inpatient and outpatient services? Summarize your finding.
- Do Some of the same patients receive both inpatient and outpatient services?  Summarize your finding.


In [35]:
train_inpatient.head(3)

BeneID   ClaimID ClaimStartDt  ClaimEndDt  Provider  \
0  BENE11001  CLM46614   2009-04-12  2009-04-18  PRV55912   
1  BENE11001  CLM66048   2009-08-31  2009-09-02  PRV55907   
2  BENE11001  CLM68358   2009-09-17  2009-09-20  PRV56046   

   InscClaimAmtReimbursed AttendingPhysician OperatingPhysician  \
0                   26000          PHY390922                NaN   
1                    5000          PHY318495          PHY318495   
2                    5000          PHY372395                NaN   

  OtherPhysician AdmissionDt  ... ClmDiagnosisCode_7  ClmDiagnosisCode_8  \
0            NaN  2009-04-12  ...               2724               19889   
1            NaN  2009-08-31  ...                NaN                 NaN   
2      PHY324689  2009-09-17  ...                NaN                 NaN   

  ClmDiagnosisCode_9 ClmDiagnosisCode_10 ClmProcedureCode_1  \
0               5849                 NaN                NaN   
1                NaN                 NaN             7092.0   
2                NaN                 NaN                NaN   

  ClmProcedureCode_2 ClmProcedureCode_3 ClmProcedureCode_4 ClmProcedureCode_5  \
0                NaN                NaN                NaN                NaN   
1                NaN                NaN                NaN                NaN   
2                NaN                NaN                NaN                NaN   

  ClmProcedureCode_6  
0                NaN  
1                NaN  
2                NaN  

[3 rows x 30 columns]

In [36]:
train_outpatient.head(3)

BeneID    ClaimID ClaimStartDt  ClaimEndDt  Provider  \
0  BENE11002  CLM624349   2009-10-11  2009-10-11  PRV56011   
1  BENE11003  CLM189947   2009-02-12  2009-02-12  PRV57610   
2  BENE11003  CLM438021   2009-06-27  2009-06-27  PRV57595   

   InscClaimAmtReimbursed AttendingPhysician OperatingPhysician  \
0                      30          PHY326117                NaN   
1                      80          PHY362868                NaN   
2                      10          PHY328821                NaN   

  OtherPhysician ClmDiagnosisCode_1  ... ClmDiagnosisCode_9  \
0            NaN              78943  ...                NaN   
1            NaN               6115  ...                NaN   
2            NaN               2723  ...                NaN   

  ClmDiagnosisCode_10 ClmProcedureCode_1 ClmProcedureCode_2  \
0                 NaN                NaN                NaN   
1                 NaN                NaN                NaN   
2                 NaN                NaN                NaN   

  ClmProcedureCode_3 ClmProcedureCode_4 ClmProcedureCode_5 ClmProcedureCode_6  \
0                NaN                NaN                NaN                NaN   
1                NaN                NaN                NaN                NaN   
2                NaN                NaN                NaN                NaN   

  DeductibleAmtPaid  ClmAdmitDiagnosisCode  
0                 0                  56409  
1                 0                  79380  
2                 0                    NaN  

[3 rows x 27 columns]

In [38]:
train_outpatient.OperatingPhysician.value_counts()

PHY330576    424
PHY424897    293
PHY314027    256
PHY423534    250
PHY357120    249
            ... 
PHY331732      1
PHY322613      1
PHY387932      1
PHY426425      1
PHY396509      1
Name: OperatingPhysician, Length: 28532, dtype: int64

In [45]:
train_outpatient.OperatingPhysician.nunique()

28532

In [47]:
#How many medical doctors are there in the train outpatient dataset?
train_outpatient.AttendingPhysician.value_counts()

PHY330576    2534
PHY350277    1628
PHY412132    1321
PHY423534    1223
PHY314027    1200
             ... 
PHY406411       1
PHY410520       1
PHY347930       1
PHY319191       1
PHY419819       1
Name: AttendingPhysician, Length: 74109, dtype: int64

In [48]:
train_outpatient.AttendingPhysician.nunique()

74109

In [49]:
train_outpatient.OtherPhysician.nunique()

44388

In [58]:
#How many medical doctors are there in the train outpatient dataset?
column_values = train_outpatient[['OperatingPhysician', 'AttendingPhysician', 'OtherPhysician']].values.astype('str')
len(np.unique(column_values))

89771

In [57]:
#How many medical doctors are there in the train inpatient dataset? Do they match with those from the outpatient 
#record?
column_values1 = train_inpatient[['OperatingPhysician', 'AttendingPhysician', 'OtherPhysician']].values.astype('str')
len(np.unique(column_values1))

18257

In [59]:
outpatient_docters = np.unique(column_values)
inpatient_docters = np.unique(column_values1)

In [62]:
outpatient_docters

array(['PHY311001', 'PHY311003', 'PHY311004', ..., 'PHY434157',
       'PHY434158', 'nan'], dtype='<U9')

In [65]:
len(np.intersect1d(outpatient_docters, inpatient_docters))

7290

7290 docters are common between the inpatient and outpatient files

## Merging Dataset

In [66]:
#merge inpatient and outpatient
#create column whether patient is outpatient(1) or not(0)
train_inpatient['isoutpatient'] = 0
train_outpatient['isoutpatient'] = 1


In [314]:
train_patient = pd.concat([train_inpatient, train_outpatient], axis = 0)

In [315]:
train_patient.head(3)

BeneID   ClaimID ClaimStartDt  ClaimEndDt  Provider  \
0  BENE11001  CLM46614   2009-04-12  2009-04-18  PRV55912   
1  BENE11001  CLM66048   2009-08-31  2009-09-02  PRV55907   
2  BENE11001  CLM68358   2009-09-17  2009-09-20  PRV56046   

   InscClaimAmtReimbursed AttendingPhysician OperatingPhysician  \
0                   26000          PHY390922                NaN   
1                    5000          PHY318495          PHY318495   
2                    5000          PHY372395                NaN   

  OtherPhysician AdmissionDt ClmAdmitDiagnosisCode  DeductibleAmtPaid  \
0            NaN  2009-04-12                  7866             1068.0   
1            NaN  2009-08-31                  6186             1068.0   
2      PHY324689  2009-09-17                 29590             1068.0   

  DischargeDt DiagnosisGroupCode ClmDiagnosisCode_1 ClmDiagnosisCode_2  \
0  2009-04-18                201               1970               4019   
1  2009-09-02                750               6186               2948   
2  2009-09-20                883              29623              30390   

  ClmDiagnosisCode_3 ClmDiagnosisCode_4 ClmDiagnosisCode_5 ClmDiagnosisCode_6  \
0               5853               7843               2768              71590   
1              56400                NaN                NaN                NaN   
2              71690              34590              V1581              32723   

  ClmDiagnosisCode_7 ClmDiagnosisCode_8 ClmDiagnosisCode_9  \
0               2724              19889               5849   
1                NaN                NaN                NaN   
2                NaN                NaN                NaN   

  ClmDiagnosisCode_10  ClmProcedureCode_1  ClmProcedureCode_2  \
0                 NaN                 NaN                 NaN   
1                 NaN              7092.0                 NaN   
2                 NaN                 NaN                 NaN   

   ClmProcedureCode_3  ClmProcedureCode_4  ClmProcedureCode_5  \
0                 NaN                 NaN                 NaN   
1                 NaN                 NaN                 NaN   
2                 NaN                 NaN                 NaN   

   ClmProcedureCode_6  isoutpatient  
0                 NaN             0  
1                 NaN             0  
2                 NaN             0

In [316]:
train_patient.tail(3)

BeneID    ClaimID ClaimStartDt  ClaimEndDt  Provider  \
517734  BENE159198  CLM596444   2009-09-24  2009-09-24  PRV53676   
517735  BENE159198  CLM636992   2009-10-18  2009-10-18  PRV53689   
517736  BENE159198  CLM686139   2009-11-17  2009-11-18  PRV53689   

        InscClaimAmtReimbursed AttendingPhysician OperatingPhysician  \
517734                      60          PHY361063                NaN   
517735                      70          PHY403198                NaN   
517736                      80          PHY419379                NaN   

       OtherPhysician AdmissionDt ClmAdmitDiagnosisCode  DeductibleAmtPaid  \
517734            NaN         NaN                   NaN                0.0   
517735      PHY419379         NaN                   NaN                0.0   
517736      PHY419379         NaN                   NaN                0.0   

       DischargeDt DiagnosisGroupCode ClmDiagnosisCode_1 ClmDiagnosisCode_2  \
517734         NaN                NaN               V570              78079   
517735         NaN                NaN                NaN                NaN   
517736         NaN                NaN              78900              78609   

       ClmDiagnosisCode_3 ClmDiagnosisCode_4 ClmDiagnosisCode_5  \
517734                NaN                NaN                NaN   
517735                NaN                NaN                NaN   
517736               4280              71946               3310   

       ClmDiagnosisCode_6 ClmDiagnosisCode_7 ClmDiagnosisCode_8  \
517734                NaN                NaN                NaN   
517735                NaN                NaN                NaN   
517736              75311               2724               V103   

       ClmDiagnosisCode_9 ClmDiagnosisCode_10  ClmProcedureCode_1  \
517734                NaN                 NaN                 NaN   
517735                NaN                 NaN                 NaN   
517736                NaN                 NaN                 NaN   

        ClmProcedureCode_2  ClmProcedureCode_3  ClmProcedureCode_4  \
517734                 NaN                 NaN                 NaN   
517735                 NaN                 NaN                 NaN   
517736                 NaN                 NaN                 NaN   

        ClmProcedureCode_5  ClmProcedureCode_6  isoutpatient  
517734                 NaN                 NaN             1  
517735                 NaN                 NaN             1  
517736                 NaN                 NaN             1

In [317]:
train_patient.shape

(558211, 31)

In [318]:
train_patient2 = train_patient.merge(train_bene, how = 'left', on = 'BeneID')

In [319]:
patient_record = train_patient2.merge(train_provider, how = 'left', on = 'Provider')

In [320]:
print('The final patient train record contains', patient_record.shape)

The final patient train record contains (558211, 56)


In [93]:
column_values = patient_record[['OperatingPhysician', 'AttendingPhysician', 'OtherPhysician']].values.astype('str')

In [88]:
len(np.unique(column_values))

100738

In [118]:
patient_record.head(3)

BeneID   ClaimID ClaimStartDt  ClaimEndDt  Provider  \
0  BENE11001  CLM46614   2009-04-12  2009-04-18  PRV55912   
1  BENE11001  CLM66048   2009-08-31  2009-09-02  PRV55907   
2  BENE11001  CLM68358   2009-09-17  2009-09-20  PRV56046   

   InscClaimAmtReimbursed AttendingPhysician OperatingPhysician  \
0                   26000          PHY390922                NaN   
1                    5000          PHY318495          PHY318495   
2                    5000          PHY372395                NaN   

  OtherPhysician AdmissionDt ClmAdmitDiagnosisCode  DeductibleAmtPaid  \
0            NaN  2009-04-12                  7866             1068.0   
1            NaN  2009-08-31                  6186             1068.0   
2      PHY324689  2009-09-17                 29590             1068.0   

  DischargeDt DiagnosisGroupCode ClmDiagnosisCode_1 ClmDiagnosisCode_2  \
0  2009-04-18                201               1970               4019   
1  2009-09-02                750               6186               2948   
2  2009-09-20                883              29623              30390   

  ClmDiagnosisCode_3 ClmDiagnosisCode_4 ClmDiagnosisCode_5 ClmDiagnosisCode_6  \
0               5853               7843               2768              71590   
1              56400                NaN                NaN                NaN   
2              71690              34590              V1581              32723   

  ClmDiagnosisCode_7 ClmDiagnosisCode_8 ClmDiagnosisCode_9  \
0               2724              19889               5849   
1                NaN                NaN                NaN   
2                NaN                NaN                NaN   

  ClmDiagnosisCode_10  ClmProcedureCode_1  ClmProcedureCode_2  \
0                 NaN                 NaN                 NaN   
1                 NaN              7092.0                 NaN   
2                 NaN                 NaN                 NaN   

   ClmProcedureCode_3  ClmProcedureCode_4  ClmProcedureCode_5  \
0                 NaN                 NaN                 NaN   
1                 NaN                 NaN                 NaN   
2                 NaN                 NaN                 NaN   

   ClmProcedureCode_6  isoutpatient         DOB  DOD  Gender  Race  \
0                 NaN             0  1943-01-01  NaN       1     1   
1                 NaN             0  1943-01-01  NaN       1     1   
2                 NaN             0  1943-01-01  NaN       1     1   

  RenalDiseaseIndicator  State  County  NoOfMonths_PartACov  \
0                     0     39     230                   12   
1                     0     39     230                   12   
2                     0     39     230                   12   

   NoOfMonths_PartBCov  ChronicCond_Alzheimer  ChronicCond_Heartfailure  \
0                   12                      1                         2   
1                   12                      1                         2   
2                   12                      1                         2   

   ChronicCond_KidneyDisease  ChronicCond_Cancer  ChronicCond_ObstrPulmonary  \
0                          1                   2                           2   
1                          1                   2                           2   
2                          1                   2                           2   

   ChronicCond_Depression  ChronicCond_Diabetes  ChronicCond_IschemicHeart  \
0                       1                     1                          1   
1                       1                     1                          1   
2                       1                     1                          1   

   ChronicCond_Osteoporasis  ChronicCond_rheumatoidarthritis  \
0                         2                                1   
1                         2                                1   
2                         2                                1   

   ChronicCond_stroke  IPAnnualReimbursementAmt  IPAnnualDeductibleAmt  \
0     

100738 unique docters are present in inpatient and outpatient

In [95]:
patient_record.dtypes

BeneID                              object
ClaimID                             object
ClaimStartDt                        object
ClaimEndDt                          object
Provider                            object
InscClaimAmtReimbursed               int64
AttendingPhysician                  object
OperatingPhysician                  object
OtherPhysician                      object
AdmissionDt                         object
ClmAdmitDiagnosisCode               object
DeductibleAmtPaid                  float64
DischargeDt                         object
DiagnosisGroupCode                  object
ClmDiagnosisCode_1                  object
ClmDiagnosisCode_2                  object
ClmDiagnosisCode_3                  object
ClmDiagnosisCode_4                  object
ClmDiagnosisCode_5                  object
ClmDiagnosisCode_6                  object
ClmDiagnosisCode_7                  object
ClmDiagnosisCode_8                  object
ClmDiagnosisCode_9                  object
ClmDiagnosi

In [190]:
#Change the chronic condition to 1 or 0 encoding from 1, 2

patient_record = patient_record.replace({'ChronicCond_Alzheimer' :2, 'ChronicCond_Heartfailure' : 2, 
                                         'ChronicCond_KidneyDisease' :2,'ChronicCond_Cancer' :2, 
                                         'ChronicCond_ObstrPulmonary' :2 ,'ChronicCond_Depression' :2,
                                        'ChronicCond_Diabetes' :2, 'ChronicCond_IschemicHeart' :2,
                                         'ChronicCond_Osteoporasis' :2, 'ChronicCond_rheumatoidarthritis' :2,
                                        'ChronicCond_stroke' :2}, 0)


In [192]:
#change the renal disease indicator in binary encoding(0,1)
patient_record = patient_record.replace({'RenalDiseaseIndicator' :'Y'},1 )

In [241]:
#Do those inpatient patients show worse health conditions (in terms of chronic diseases) than typical outpatient 
#patients, or do those who have more visits to the providers have worse health conditions? Provide an analysis 
#on these issues.

patient_record[patient_record['ChronicCond_Heartfailure'] == 1].groupby('isoutpatient').size()

isoutpatient
0     29215
1    300368
dtype: int64

In [242]:
print(29215/40474)#percentage of inpatient with chronic condition
print(300368/517737)#percentage of outpatient with chronic condition

0.7218214162178188
0.5801555616075343


In [243]:
patient_record[patient_record['ChronicCond_Alzheimer'] == 1].groupby('isoutpatient').size()

isoutpatient
0     20401
1    203926
dtype: int64

In [244]:
print(20401/40474)#percentage of inpatient with chronic condition
print(203926/517737)#percentage of outpatient with chronic condition

0.5040519839897218
0.3938795179792057


In [237]:
patient_record[patient_record['ChronicCond_KidneyDisease'] == 1].groupby('isoutpatient').size()

isoutpatient
0     24356
1    205628
dtype: int64

In [238]:
print(24356/40474)#percentage of inpatient with chronic condition
print(205628/517737)#percentage of outpatient with chronic condition

0.6017690369125859
0.39716690134180094


In [233]:
patient_record[patient_record['ChronicCond_Cancer'] == 1].groupby('isoutpatient').size()

isoutpatient
0     7729
1    76776
dtype: int64

In [234]:
print(7729/40474)#percentage of inpatient with chronic condition
print(76776/517737)#percentage of outpatient with chronic condition

0.19096209912536444
0.14829150707791794


In [231]:
patient_record[patient_record['ChronicCond_ObstrPulmonary'] == 1].groupby('isoutpatient').size()

isoutpatient
0     19393
1    155288
dtype: int64

In [232]:
print(19393/40474)#percentage of inpatient with chronic condition
print(155288/517737)#percentage of outpatient with chronic condition

0.4791471067846025
0.2999360679263796


In [225]:
patient_record[patient_record['ChronicCond_Depression'] == 1].groupby('isoutpatient').size()

isoutpatient
0     20327
1    222387
dtype: int64

In [226]:
print(20327/40474)#percentage of inpatient with chronic condition
print(222387/517737)#percentage of outpatient with chronic condition

0.5022236497504571
0.42953661801262033


In [220]:
patient_record[patient_record['ChronicCond_Diabetes'] == 1].groupby('isoutpatient').size()

isoutpatient
0     32462
1    361297
dtype: int64

In [224]:
print(32462/40474)#percentage of inpatient with chronic condition
print(361297/517737)#percentage of outpatient with chronic condition

0.8020457577704205
0.6978388641337204


In [215]:
patient_record[patient_record['ChronicCond_IschemicHeart'] == 1].groupby('isoutpatient').size()

isoutpatient
0     34674
1    389156
dtype: int64

In [217]:
print(34674/40474)#percentage of inpatient with chronic condition
print(389156/517737)#percentage of outpatient with chronic condition

0.8566981271927657
0.7516480375171177


In [203]:
patient_record[patient_record['ChronicCond_Osteoporasis'] == 1].groupby('isoutpatient').size()

isoutpatient
0     14229
1    163085
dtype: int64

In [206]:
print(14229/40474)#percentage of inpatient with chronic condition
print(163085/517737)#percentage of outpatient with chronic condition

0.3515590255472649
0.31499583765502565


In [211]:
patient_record[patient_record['ChronicCond_stroke'] == 1].groupby('isoutpatient').size()

isoutpatient
0     6874
1    49907
dtype: int64

In [212]:
print(6874/40474)#percentage of inpatient provider with chronic condition
print(49907/517737)#percentage of outpatient with chronic condition

0.16983742649602213
0.09639450145537212


In [213]:
patient_record[patient_record['ChronicCond_rheumatoidarthritis'] == 1].groupby('isoutpatient').size()

isoutpatient
0     14585
1    159114
dtype: int64

In [214]:
print(14585/40474)#percentage of inpatient with chronic condition
print(159114/517737)#percentage of outpatient with chronic condition

0.36035479567129514
0.3073259203031655


In [247]:
patient_record.groupby('ChronicCond_Heartfailure')['Provider'].count()

ChronicCond_Heartfailure
0    228628
1    329583
Name: Provider, dtype: int64

In [248]:
#whether the patient with chronic disease visits the provider most of the time
print(228628/558211)#percentage of patient with no chronic condition visiting provider
print(329583/558211)#percentage of patient with chronic condition visiting provider

0.4095727242924271
0.5904272757075729


In [245]:
patient_record.groupby('ChronicCond_Alzheimer')['Provider'].count()

ChronicCond_Alzheimer
0    333884
1    224327
Name: Provider, dtype: int64

In [246]:
#whether the patient with chronic disease visits the provider most of the time
print(333884/558211)#percentage of patient with no chronic condition visiting provider
print(224327/558211)#percentage of patient with chronic condition visiting provider

0.5981322474834785
0.4018677525165215


In [239]:
patient_record.groupby('ChronicCond_KidneyDisease')['Provider'].count()

ChronicCond_KidneyDisease
0    328227
1    229984
Name: Provider, dtype: int64

In [240]:
#whether the patient with chronic disease visits the provider most of the time
print(328227/558211)#percentage of patient with no chronic condition visiting provider
print(229984/558211)#percentage of patient with chronic condition visiting provider

0.5879980867449763
0.4120019132550236


In [235]:
patient_record.groupby('ChronicCond_Cancer')['Provider'].count()

ChronicCond_Cancer
0    473706
1     84505
Name: Provider, dtype: int64

In [236]:
#whether the patient with chronic disease visits the provider most of the time
print(473706/558211)#percentage of patient with no chronic condition visiting provider
print(84505/558211)#percentage of patient with chronic condition visiting provider

0.848614591973286
0.1513854080267139


In [229]:
patient_record.groupby('ChronicCond_ObstrPulmonary')['Provider'].count()

ChronicCond_ObstrPulmonary
0    383530
1    174681
Name: Provider, dtype: int64

In [230]:
#whether the patient with chronic disease visits the provider most of the time
print(383530/558211)#percentage of patient with no chronic condition visiting provider
print(174681/558211)#percentage of patient with chronic condition visiting provider

0.6870699430860373
0.31293005691396264


In [227]:
patient_record.groupby('ChronicCond_Depression')['Provider'].count()

ChronicCond_Depression
0    315497
1    242714
Name: Provider, dtype: int64

In [228]:
#whether the patient with chronic disease visits the provider most of the time
print(315497/558211)#percentage of patient with no chronic condition visiting provider
print(242714/558211)#percentage of patient with chronic condition visiting provider

0.5651930900680925
0.43480690993190746


In [222]:
patient_record.groupby('ChronicCond_Diabetes')['Provider'].count()

ChronicCond_Diabetes
0    164452
1    393759
Name: Provider, dtype: int64

In [223]:
#whether the patient with chronic disease visits the provider most of the time
print(164452/558211)#percentage of patient with no chronic condition visiting provider
print(393759/558211)#percentage of patient with chronic condition visiting provider

0.2946054448944933
0.7053945551055067


In [218]:
patient_record.groupby('ChronicCond_IschemicHeart')['Provider'].count()

ChronicCond_IschemicHeart
0    134381
1    423830
Name: Provider, dtype: int64

In [219]:
#whether the patient with chronic disease visits the provider most of the time
print(134381/558211)#percentage of patient with no chronic condition visiting provider
print(423830/558211)#percentage of patient with chronic condition visiting provider

0.24073513420552445
0.7592648657944756


In [204]:
patient_record.groupby('ChronicCond_Osteoporasis')['Provider'].count()

ChronicCond_Osteoporasis
0    380897
1    177314
Name: Provider, dtype: int64

In [207]:
#whether the patient with chronic disease visits the provider most of the time
print(380897/558211)#percentage of patient with no chronic condition visiting provider
print(177314/558211)#percentage of patient with chronic condition visiting provider

0.6823530887066002
0.3176469112933998


In [210]:
patient_record.groupby('ChronicCond_stroke')['Provider'].count()

ChronicCond_stroke
0    501430
1     56781
Name: Provider, dtype: int64

In [249]:
#whether the patient with chronic disease visits the provider most of the time
print(501430/558211)#percentage of patient with no chronic condition visiting provider
print(56781/558211)#percentage of patient with chronic condition visiting provider

0.8982803993472003
0.10171960065279975


In [188]:
patient_record.groupby('ChronicCond_rheumatoidarthritis')['Provider'].count()

ChronicCond_rheumatoidarthritis
1    173699
2    384512
Name: Provider, dtype: int64

In [179]:
#whether the patient with chronic disease visits the provider most of the time
print(173699/558211)#percentage of patient with no chronic condition visiting provider
print(384512/558211)#percentage of patient with chronic condition visiting provider

0.311170865497097
0.6888291345029031


The inpatients are have more chronic condition. The number of times a patient with chronic condition visits more provider.

In [251]:
#How many distinct patients (BeneIDs) are there in the in/out-patient datasets?
patient_record.groupby('isoutpatient')['BeneID'].nunique()

isoutpatient
0     31289
1    133980
Name: BeneID, dtype: int64

The inpatients have 31289 distinct beneficiary and outpatient have 133980 distinct beneficiary

In [ ]:
#Do doctors serve for different providers? Study the distribution of hospital counts/doctor? Is it possible to 
#characterize those doctors who move around among different providers?

In [254]:
column_values = patient_record[['OperatingPhysician', 'AttendingPhysician', 'OtherPhysician']].values.astype('str')

In [326]:
multiple_phy = patient_record.groupby('AttendingPhysician')['Provider'].count().reset_index()

In [327]:
multiple_phy.head(2)

AttendingPhysician  Provider
0          PHY311001         2
1          PHY311002         1

In [328]:
multiphy_list = multiple_phy [multiple_phy.Provider > 1].AttendingPhysician.to_list()

In [342]:
grp1 = patient_record[patient_record['AttendingPhysician'].isin(multiphy_list)].groupby(['isoutpatient', 'PotentialFraud'])
grp1['PotentialFraud'].count()

isoutpatient  PotentialFraud
0             No                 13520
              Yes                21977
1             No                302877
              Yes               182230
Name: PotentialFraud, dtype: int64

In [340]:
str('AttendingPhysician') + 'count'

'AttendingPhysiciancount'

In [334]:
patient_record[patient_record['AttendingPhysician'].isin(multiphy_list)].shape

(520604, 56)

In [275]:
chronic = patient_record.columns[patient_record.columns.str.contains("Chronic")].tolist()

In [285]:
claims = patient_record.groupby('BeneID').count().ClaimID.reset_index()


In [290]:
bene_list = claims[claims.ClaimID > 1].BeneID.to_list()

In [297]:
patient_record[patient_record['BeneID'].isin(bene_list)].shape

(525337, 56)

In [301]:
#Patient with chronic disease visiting more than 1
patient_record[patient_record['BeneID'].isin(bene_list)][chronic].sum()/patient_record[patient_record['BeneID'].isin(bene_list)].shape[0]

ChronicCond_Alzheimer              0.412986
ChronicCond_Heartfailure           0.606022
ChronicCond_KidneyDisease          0.426724
ChronicCond_Cancer                 0.156361
ChronicCond_ObstrPulmonary         0.324514
ChronicCond_Depression             0.447159
ChronicCond_Diabetes               0.723132
ChronicCond_IschemicHeart          0.774370
ChronicCond_Osteoporasis           0.325096
ChronicCond_rheumatoidarthritis    0.320488
ChronicCond_stroke                 0.105220
dtype: float64

In [302]:
#Patient with chronic disease visiting once
patient_record[~patient_record['BeneID'].isin(bene_list)][chronic].sum()/patient_record[~patient_record['BeneID'].isin(bene_list)].shape[0]

ChronicCond_Alzheimer              0.224189
ChronicCond_Heartfailure           0.341212
ChronicCond_KidneyDisease          0.176735
ChronicCond_Cancer                 0.071881
ChronicCond_ObstrPulmonary         0.127821
ChronicCond_Depression             0.237422
ChronicCond_Diabetes               0.421944
ChronicCond_IschemicHeart          0.517886
ChronicCond_Osteoporasis           0.198607
ChronicCond_rheumatoidarthritis    0.162286
ChronicCond_stroke                 0.045781
dtype: float64

In [ ]:
patient[patient["BeneID"].isin(bene_more)][chronic].sum()/(patient[patient["BeneID"].isin(bene_more)].shape[0])

In [ ]:
num_clm = patient.groupby('BeneID').count().ClaimID.reset_index()

In [303]:
nuniq_prov = patient_record.groupby('AttendingPhysician')["Provider"].nunique().reset_index()
phy_more = nuniq_prov[nuniq_prov.Provider > 1].AttendingPhysician.tolist()

In [307]:
patient_record.loc[patient_record["AttendingPhysician"].isin(phy_more),"Att_Phy_Mult"] = 1

In [312]:
patient_record.columns

Index(['BeneID', 'ClaimID', 'ClaimStartDt', 'ClaimEndDt', 'Provider',
       'InscClaimAmtReimbursed', 'AttendingPhysician', 'OperatingPhysician',
       'OtherPhysician', 'AdmissionDt', 'ClmAdmitDiagnosisCode',
       'DeductibleAmtPaid', 'DischargeDt', 'DiagnosisGroupCode',
       'ClmDiagnosisCode_1', 'ClmDiagnosisCode_2', 'ClmDiagnosisCode_3',
       'ClmDiagnosisCode_4', 'ClmDiagnosisCode_5', 'ClmDiagnosisCode_6',
       'ClmDiagnosisCode_7', 'ClmDiagnosisCode_8', 'ClmDiagnosisCode_9',
       'ClmDiagnosisCode_10', 'ClmProcedureCode_1', 'ClmProcedureCode_2',
       'ClmProcedureCode_3', 'ClmProcedureCode_4', 'ClmProcedureCode_5',
       'ClmProcedureCode_6', 'isoutpatient', 'DOB', 'DOD', 'Gender', 'Race',
       'RenalDiseaseIndicator', 'State', 'County', 'NoOfMonths_PartACov',
       'NoOfMonths_PartBCov', 'ChronicCond_Alzheimer',
       'ChronicCond_Heartfailure', 'ChronicCond_KidneyDisease',
       'ChronicCond_Cancer', 'ChronicCond_ObstrPulmonary',
       'ChronicCond_Depressi

In [321]:
patient_record.shape

(558211, 56)